File: example/src/main/java/eu/darken/mvpbakery/example/screens/counting/CountingComponent.java
package eu.darken.mvpbakery.example.screens.counting;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import dagger.Subcomponent;
import eu.darken.mvpbakery.injection.PresenterComponent;
import eu.darken.mvpbakery.injection.fragment.FragmentComponent;

@CountingComponent.Scope
@Subcomponent(modules = {CountingModule.class})
public interface CountingComponent extends FragmentComponent<CountingFragment>, PresenterComponent<CountingPresenter.View, CountingPresenter> {

    @Subcomponent.Builder
    abstract class Builder extends FragmentComponent.Builder<CountingFragment, CountingComponent> {
        public abstract Builder module(CountingModule module);
    }

    @javax.inject.Scope
    @Retention(RetentionPolicy.RUNTIME)
    @interface Scope {
    }
}

File: example/src/main/java/eu/darken/mvpbakery/example/screens/counting/CountingComponent.kt
package eu.darken.mvpbakery.example.screens.counting

import dagger.Subcomponent
import eu.darken.mvpbakery.injection.PresenterComponent
import eu.darken.mvpbakery.injection.fragment.FragmentComponent

@CountingComponent.Scope
@Subcomponent(modules = [CountingModule::class])
interface CountingComponent : FragmentComponent<CountingFragment>, PresenterComponent<CountingPresenter, CountingComponent> {

    @Subcomponent.Builder
    abstract class Builder : FragmentComponent.Builder<CountingFragment, CountingComponent>() {
        abstract fun module(module: CountingModule): Builder
    }

    @javax.inject.Scope
    @Retention(AnnotationRetention.RUNTIME)
    annotation class Scope
}
